fastcrypto/src/tests/secp256r1_tests.rs

```diff
@@ -0,0 +1,446 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use p256::ecdsa::Signature;
+//#[cfg(feature = "copy_key")]
+//use proptest::arbitrary::Arbitrary;
+use rand::{rngs::StdRng, SeedableRng as _};
+use rust_secp256k1::constants::SECRET_KEY_SIZE;
+
+use signature::{Signer, Verifier};
+use wycheproof::ecdsa::{TestName::EcdsaSecp256r1Sha256, TestSet};
+use wycheproof::TestResult;
+
+use crate::{
+    hash::{HashFunction, Sha256},
+    secp256r1::{
+        Secp256r1KeyPair, Secp256r1PrivateKey, Secp256r1PublicKey, Secp256r1PublicKeyBytes,
+        Secp256r1Signature,
+    },
+    traits::{EncodeDecodeBase64, KeyPair, ToFromBytes, VerifyingKey},
+};
+
+use super::*;
+
+pub fn keys() -> Vec<Secp256r1KeyPair> {
+    let mut rng = StdRng::from_seed([0; 32]);
+
+    (0..4)
+        .map(|_| Secp256r1KeyPair::generate(&mut rng))
+        .collect()
+}
+
+#[test]
+fn serialize_deserialize() {
+    let kpref = keys().pop().unwrap();
+    let public_key = kpref.public();
+
+    let bytes = bincode::serialize(&public_key).unwrap();
+    let pk2 = bincode::deserialize::<Secp256r1PublicKey>(&bytes).unwrap();
+    assert_eq!(public_key.as_ref(), pk2.as_ref());
+
+    let private_key = kpref.private();
+    let bytes = bincode::serialize(&private_key).unwrap();
+    let privkey = bincode::deserialize::<Secp256r1PrivateKey>(&bytes).unwrap();
+    let bytes2 = bincode::serialize(&privkey).unwrap();
+    assert_eq!(bytes, bytes2);
+
+    let signature = Secp256r1Signature::default();
+    let bytes = bincode::serialize(&signature).unwrap();
+    let sig = bincode::deserialize::<Secp256r1Signature>(&bytes).unwrap();
+    let bytes2 = bincode::serialize(&sig).unwrap();
+    assert_eq!(bytes, bytes2);
+
+    // test serde_json serialization
+    let serialized = serde_json::to_string(&signature).unwrap();
+    println!("{:?}", serialized);
+    let deserialized: Secp256r1Signature = serde_json::from_str(&serialized).unwrap();
+    assert_eq!(deserialized.as_ref(), signature.as_ref());
+}
+
+#[test]
+fn import_export_public_key() {
+    let kpref = keys().pop().unwrap();
+    let public_key = kpref.public();
+    let export = public_key.encode_base64();
+    let import = Secp256r1PublicKey::decode_base64(&export);
+    assert!(import.is_ok());
+    assert_eq!(import.unwrap().as_ref(), public_key.as_ref());
+}
+
+#[test]
+fn test_public_key_bytes_conversion() {
+    let kp = keys().pop().unwrap();
+    let pk_bytes: Secp256r1PublicKeyBytes = kp.public().into();
+    let rebuilt_pk: Secp256r1PublicKey = pk_bytes.try_into().unwrap();
+    assert_eq!(kp.public().as_bytes(), rebuilt_pk.as_bytes());
+}
+
+#[test]
+fn test_public_key_recovery() {
+    let kp = keys().pop().unwrap();
+    let message: &[u8] = b"Hello, world!";
+    let signature: Secp256r1Signature = kp.sign(message);
+    let recovered_keys = signature.recover(message).unwrap();
+    assert!(recovered_keys.contains(kp.public()));
+}
+//
+// #[test]
+// fn test_public_key_recovery_error() {
```

Comment:
It would be great to at least add commentary on what this test should become if we don't have the time to fix it.